### PR TITLE
37003 - Send email accepted request

### PR DESCRIPTION
**Pull request by vinhnd2010**

### CODE REVIEW

File: app/mailers/application_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class ApplicationMailer < ActionMailer::Base
+  default from: "from@example.com"
+  layout 'mailer'
```
Comment (trinhducduy) [inline app/mailers/application_mailer.rb]: ' to "

File: config/environments/development.rb [hunk 2]
```diff
@@ -15,7 +15,18 @@
 
   # Don't care if the mailer can't send.
   config.action_mailer.raise_delivery_errors = false
-
+  config.action_mailer.delivery_method = :smtp
+  host = "localhost:3000"
+  config.action_mailer.default_url_options = {host: host}
+
+  ActionMailer::Base.smtp_settings = {
```
Comment (trinhducduy) [inline config/environments/development.rb]: ' to "

File: app/views/request_mailer/acception_request.text.erb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Hi <%= @user.name %>,
+
+Congratulation on your accepted book request.
+You can see your request at:
+<%= link_to t("t.subject.request.see_request"), request_path(@request, email: @user.email) %>
```
Comment (trinhducduy) [inline app/views/request_mailer/acception_request.text.erb]: remove last ()

File: app/views/request_mailer/acception_request.html.erb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<h1><%= t "email.subject.request" %></h1>
+<p>
+  Hi <%= @user.name %>,
+  Congratulation on your accepted book request.
+  You can see your request at:
+</p>
+<%= link_to t("t.subject.request.see_request"), request_path(@request) %>
```
Comment (trinhducduy) [inline app/views/request_mailer/acception_request.html.erb]: remove last ()

File: config/environments/development.rb [hunk 5]
```diff
@@ -13,9 +13,20 @@
...
+  # Don"t care if the mailer can"t send.
   config.action_mailer.raise_delivery_errors = false
-
+  config.action_mailer.delivery_method = :smtp
+  host = "localhost:3000"
+  config.action_mailer.default_url_options = {host: host}
+
+  ActionMailer::Base.smtp_settings = {
+    :address        => "smtp.gmail.com",
+    :port           => "587",
```
Comment (trinhducduy) [inline config/environments/development.rb]: convention

File: config/environments/development.rb [hunk 6]
```diff
@@ -1,9 +1,9 @@
 Rails.application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
```
Comment (ThuBM) [inline config/environments/development.rb]: remove all commented lines

File: config/routes.rb [hunk 7]
```diff
@@ -12,7 +12,7 @@
   end
   resources :reviews, except: [:index, :show]
   resources :comments, except: [:index, :show]
-  resources :requests, only: [:create, :destroy, :index, :new]
+  resources :requests, only: [:create, :destroy, :index, :new, :show]
```
Comment (ThuBM) [inline config/routes.rb]: except

**Comment by ThuBM** [comment]
ok

File: app/views/request_mailer/acception_request.html.erb [hunk 8]
```diff
@@ -0,0 +1,8 @@
+<h1><%= t "email.subject.request" %></h1>
+<p>
+  Hi <%= @user.name %>,
+  Congratulation on your accepted book request.
+  You can see your request at:
+</p>
+<%= link_to t("t.subject.request.see_request"),
+  "http://localhost:3000/#{requests_path}" %>
```
Comment (trinhducduy) [inline app/views/request_mailer/acception_request.html.erb]: no need http://localhost:3000

File: app/controllers/admin/requests_controller.rb [hunk 9]
```diff
@@ -6,6 +6,7 @@ def index
   def update
     @request = Request.find params[:id]
     if @request.update_attributes request_params
+      RequestMailer.acception_request(@request).deliver_now
```
Comment (dieunb) [inline app/controllers/admin/requests_controller.rb]: move to call_back

File: config/environments/development.rb [hunk 10]
```diff
@@ -1,41 +1,23 @@
...
   config.eager_load = false
+  config.consider_all_requests_local = true
+
+  config.action_mailer.raise_delivery_errors = true
+  config.action_mailer.default_url_options = {host: "localhost", port: 3000}
+  config.action_mailer.delivery_method = :smtp
+  config.action_mailer.smtp_settings = {
+    address: "smtp.gmail.com",
+    port: 587,
+    user_name: "nguyenducvinh92@gmail.com",
```
Comment (dieunb) [inline config/environments/development.rb]: lưu vào biến environment nhé

File: config/environments/development.rb [hunk 11]
```diff
@@ -1,41 +1,23 @@
...
+    password: "246810121",
```
Comment (dieunb) [inline config/environments/development.rb]: đây nữa

**Pull request merged**